crypto/pkcs11key/pkcs11key.go

```diff
@@ -30,13 +34,29 @@ var hashPrefixes = map[crypto.Hash][]byte{
 // using a key stored in a PKCS#11 hardware token.  This enables
 // the use of PKCS#11 tokens with the Go x509 library's methods
 // for signing certificates.
+//
+// Each PKCS11Key represents one session. Its session handle is
+// protected internally by a mutex, so at most one Sign operation can be active
+// at a time. For best performance you may want to instantiate multiple PKCS11Keys.
+// Each one will have its own session and can be used concurrently. Note that
+// some smartcards like the Yubikey Neo do not support multiple simultaneous
+// sessions and will error out on creation of the second PKCS11Key object.
+//
+// Note: For parallel usage, it is *highly* recommended that you create all your
+// PKCS11Key objects serially, on your main thread, checking for errors each
+// time, and then farm them out for use by different goroutines. If you fail to
+// do this, your application may attempt to login repeatedly with an incorrect
+// PIN, locking the PKCS#11 token.
 type PKCS11Key struct {
```

Comment:
One of the common things we've heard about the PKCS #11 support is that
it's a confusing; anything we can do now to make it easier to understand
I'll support. I believe Let's Encrypt is the major PKCS #11 user, so if
these changes are productive for you all, I think that they will work
well for other people.